### PR TITLE
disable non_exhaustive if unstable_discord_api

### PR DESCRIPTION
**Pull request by cheesycod**
Disables the ``non_exhaustive`` attribute in FullEvent if ``unstable_discord_api`` is enabled as a feature. It is already expected that any change to serenity with ``unstable_discord_api`` enabled may or may not break code using this feature and making this enum exhaustive allows for use cases such as fine-grained audit logging etc. where new variants should be a compiler error

### CODE REVIEW

**Review by GnomedDev — CHANGES_REQUESTED** [review]
Please rename the feature to `unstable`, this does not make sense under the `unstable_discord_api` feature.

File: README.md
Comment (GnomedDev) [inline README.md]:
```suggestion
- **unstable**: Enables features of the Serenity and Discord API that do not have a stable interface. The features might not have official documentation and are subject to change without a breaking version bump.
```

**Review by GnomedDev — COMMENTED** [review]
Just a small nitpick about the documentation for this feature

**Review by GnomedDev — APPROVED** [review]
LGTM, but I want someone else's approval before merging

**Review by arqunis — APPROVED** [review]

**Pull request merged**
